storage/reads/datatypes/storage_common.proto

```diff
@@ -208,6 +214,22 @@ message TimestampRange {
   int64 end = 2;
 }
 
+// TagsRequest is the request message for the Storage.TagKeys and Storage.TagValues
+message TagsRequest {
+  google.protobuf.Any tags_source = 1 [(gogoproto.customname) = "TagsSource"];
+  TimestampRange range = 2 [(gogoproto.nullable) = false];
+  Predicate predicate = 3;
+}
+
+// Response message for Storage.TagKeys and Storage.TagValues.
+message TagsResponse {
+  message Frame {
+    repeated bytes values = 1;
+  }
+
+  repeated Frame frames = 1 [(gogoproto.nullable) = false];
```

Comment:
nit: I don't believe we'll need this inner `Frame` message; we can probably just include the values property in the `TagsResponse`.

Reply:
👍